multimodal/src/autogluon/multimodal/utils/environment.py

```diff
@@ -238,15 +238,14 @@ def get_precision_context(precision: Union[int, str], device_type: Optional[str]
     -------
     A precision context manager.
     """
-    if precision == 32:
+    precision = infer_precision(num_gpus=1, precision=precision, as_torch=True)
```

Comment:
hard-coded num_gpus here instead of `num_gpus = int(device_type == 'gpu')`

because I'm not sure if `get_precision_context` is supposed to support 64bit floats on cpu, and `infer_precision` disallows it.